lib/api/apiUtils/object/checkEncryption.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Checks the object encryption request against bucket encryption for matching
+ * SSE-S3 configuration
+ * @param {object} objectRequest - http request
+ * @param {object} serverSideEncryption - bucket encryption info
+ * @return {boolean} returns true if the object request has correct SSE-S3
+ * configuration
+ */
+function isValidSSES3(objectRequest, serverSideEncryption) {
+    // x-amz-server-side-encryption is allowed only if bucket
+    // encryption is enabled and if the value is AES256
+    // NOTE: object level encryption is not supported, but we allow
+    // encryption headers in the object request headers!
+    const sseHeader = objectRequest.headers['x-amz-server-side-encryption'];
+    const encryptionAlgorithm = 'AES256';
+
+    const result = ((!serverSideEncryption
+        && sseHeader === encryptionAlgorithm) ||
+        (!serverSideEncryption && sseHeader
+            && sseHeader !== encryptionAlgorithm) ||
+        (serverSideEncryption && sseHeader
```

Comment:
If bucket encryption is enabled and if its value is different from `AES256` (i.e. `rsa`), should we forbid any `x-amz-server-side-encryption` header? Maybe add more unit test with bucket encryption equals to rsa.

Reply:
We don't allow other encryption algorithms besides AES256 (from the docs I have checked it is the only one supported and I assume it is the standard). If other values are passed then encryption is not enabled but the bucket is created.